### PR TITLE
Fix plugins crash

### PR DESCRIPTION
**Pull request by Art17**
move contoller code in oracle and swap from threads to app
change websocket to http connection in swap plugin
extend http_client functionality in oracle and swap plugin
refactor code

### CODE REVIEW

File: plugins/eth_swap_plugin/eth_swap_plugin.cpp [hunk 1]
```diff
@@ -95,32 +96,42 @@ namespace eosio {
...
                     push_txs(prev_swap_requests);
                     from_block_dec = to_block_dec;
                     current_long_polling_blocks_per_filter = std::min(current_long_polling_blocks_per_filter * 2,
                                                                       long_polling_blocks_per_filter);
 
                     sleep(long_polling_period);
-                } catch (TimeoutException e) {
-                    if (current_long_polling_blocks_per_filter == 1) {
+                } /*catch (TimeoutException e) {
+                    if(current_long_polling_blocks_per_filter == 1) {
```
Comment (roman-tik) [inline plugins/eth_swap_plugin/eth_swap_plugin.cpp]: do we still need this comment here?

File: plugins/eth_swap_plugin/eth_swap_plugin.cpp [hunk 2]
```diff
@@ -95,32 +96,42 @@ namespace eosio {
...
                         elog("Eth node is not responding at block ${b}", ("b", from_block_dec));
                         sleep(wait_for_eth_node);
                     }
                     current_long_polling_blocks_per_filter /= 4;
                     current_long_polling_blocks_per_filter = std::max(1u, current_long_polling_blocks_per_filter);
-                } FC_LOG_WAIT_AND_CONTINUE()
+                } */FC_LOG_WAIT_AND_CONTINUE()
```
Comment (roman-tik) [inline plugins/eth_swap_plugin/eth_swap_plugin.cpp]: where do we decrease the number of blocks per request now?

**Review by roman-tik — CHANGES_REQUESTED** [review]

File: plugins/eth_swap_plugin/eth_swap_plugin.cpp (hunk 1 above)
Comment (Art17) [reply plugins/eth_swap_plugin/eth_swap_plugin.cpp]: Removed.

File: plugins/eth_swap_plugin/eth_swap_plugin.cpp (hunk 2 above)
Comment (Art17) [reply plugins/eth_swap_plugin/eth_swap_plugin.cpp]: No need for this functionality with http requests

File: plugins/eth_swap_plugin/eth_swap_plugin.cpp [hunk 3]
```diff
@@ -381,7 +390,7 @@ namespace eosio {
     }
 
     void eth_swap_plugin::plugin_startup() {
-        ilog("Ethereum swap plugin started. Version 2.0");
+        ilog("Ethereum swap plugin started. Version 2.0.1");
```
Comment (roman-tik) [inline plugins/eth_swap_plugin/eth_swap_plugin.cpp]: either pull version from the build or remove it completely.

**Review by roman-tik — CHANGES_REQUESTED** [review]

File: plugins/eth_swap_plugin/eth_swap_plugin.cpp (hunk 3 above)
Comment (Art17) [reply plugins/eth_swap_plugin/eth_swap_plugin.cpp]: Done.

**Pull request merged**